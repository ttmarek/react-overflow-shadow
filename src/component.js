import React, { Component } from 'react';

const styles = {
  frame: {
    overflow: 'hidden',
  },

  window: {
    position: 'relative',
    maxHeight: '100%',
    marginRight: '-50px',
    paddingRight: '50px',
    marginTop: '-50px',
    paddingTop: '50px',
  },

  overflowX: {
    overflowX: 'scroll',
    overflowY: 'hidden',
  },

  overflowY: {
    overflowX: 'hidden',
    overflowY: 'scroll',
  },


};

class OverflowShadow extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hasTopShadow: false,
      hasBottomShadow: false,
      hasLeftShadow: false,
      hasRightShadow: false,
    };
  }

  render() {
    const {
      children
    } = this.props;

    const {
      hasTopShadow,
      hasBottomShadow,
      hasLeftShadow,
      hasRightShadow,
    } = this.state;

    const topShadow = '';
    const bottomShadow = '';
    const leftShadow = '';
    const rightShadow = '';

    return (
      <div style={styles.frame}>
        <div style={styles.window}>
          {children}
          { hasTopShadow ? topShadow : null }
          { hasBottomShadow ? bottomShadow : null }
          { hasLeftShadow ? leftShadow : null }
          { hasRightShadow ? rightShadow : null }
        </div>
      </div>
    );
  }
};

export { OverflowShadow }
