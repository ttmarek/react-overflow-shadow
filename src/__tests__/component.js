import React from 'react';
import renderer from 'react-test-renderer';
import { OverflowShadow } from '../component';

describe('happy path', () => {
  it('matches the snapshot', () => {
    const component = renderer.create(
      <OverflowShadow />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
