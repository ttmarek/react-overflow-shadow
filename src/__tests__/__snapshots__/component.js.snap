exports[`happy path matches the snapshot 1`] = `
<div>
  Hello
</div>
`;
